classes/class.ilMumieTaskGradeListGUI.php

```diff
@@ -0,0 +1,207 @@
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * This form is used to add, edit and validate MUMIE Server configurations
+ * @ilCtrl_isCalledBy ilBarGUI: ilFooGUI (multiple classes can be separated by comma)
+ */
+
+require_once('./Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskGradeSync.php');
+class ilMumieTaskGradeListGUI extends ilTable2GUI
+{
+    private $parentObj;
+    private $parent_gui;
+    private $postvar;
+
+    public function __construct($parentObj)
+    {
+        global $lng;
+        $user_id = $_GET['user_id'];
+        $this->parentObj = $parentObj;
+        $this->admin_settings = ilMumieTaskAdminSettings::getInstance();
+
+        $this->setId("user" . $_GET["ref_id"]);
+        parent::__construct($parentObj, 'displayGradeList');
+
+        $this->setFormName('participants');
+
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_submission_date'), 'date');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_use_grade'), 'useGrade');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_grade'), 'grade');
+        
+        $this->tpl->addBlockFile(
+            "TBL_CONTENT",
+            "tbl_content",
+            "tpl.mumie_grade_list.html",
+            "Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask"
+        );
+        if ($parentObj->object->getPrivateGradepool() != -1) {
+            $gradesync  = new  ilMumieTaskGradeSync($parentObj->object, false);
+            $xGrades = $gradesync->getAllXapiGradesByUser();
+            $syncId = $gradesync->getSyncIds(array($user_id))[0];
+            if (!empty($xGrades)) {
+                foreach ($xGrades as $xGrade) {
+                    if ($xGrade->actor->account->name == $syncId) {
+                        $this->tpl->setCurrentBlock("tbl_content");
+                        $this->css_row = ($this->css_row != "tblrow1")
+                            ? "tblrow1"
+                            : "tblrow2";
+
+                        $this->tpl->setVariable("CSS_ROW", $this->css_row);
+                        $this->tpl->setVariable("VAL_GRADE", round($xGrade->result->score->raw * 100));
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'user_id', $user_id);
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'newGrade', round($xGrade->result->score->raw * 100));
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'timestamp', strtotime($xGrade->timestamp));
+
+                        $this->tpl->setVariable("LINK_NAME", $this->ctrl->getLinkTarget($parentObj, 'displayGradeList'));
+
+                        $this->tpl->setVariable("LINK_TXT", $lng->txt('rep_robj_xmum_frm_list_use_grade'));
+                        $this->tpl->setVariable("VAL_DATE", substr($xGrade->timestamp, 0, 10) . " - " . substr($xGrade->timestamp, 11, 5));
```

Comment:
We should not create our own logic to convert timestamps. There are pre-existing solutions to this. Have a look at `ilDateTime`